mesa/agent.py

```diff
@@ -48,25 +48,23 @@ class Agent:
     # so, unique_id is unique relative to a model, and counting starts from 1
     _ids = defaultdict(functools.partial(itertools.count, 1))
 
-    def __init__(self, model: Model, *args, **kwargs) -> None:
+    def __init__(
+        self, model: Model, *args, key_by_name: bool = False, **kwargs
+    ) -> None:
         """Create a new agent.
 
         Args:
             model (Model): The model instance in which the agent exists.
-            args: passed on to super
-            kwargs: passed on to super
-
-        Notes:
-            to make proper use of python's super, in each class remove the arguments and
-            keyword arguments you need and pass on the rest to super
-
+            key_by_name (bool): If True, use Agent.__name__ as the key in the model's agents_by_type dictionary.
+            args: Passed on to super.
+            kwargs: Passed on to super.
         """
         super().__init__(*args, **kwargs)
 
         self.model: Model = model
         self.unique_id: int = next(self._ids[model])
         self.pos: Position | None = None
-        self.model.register_agent(self)
+        self.model.register_agent(self, key_by_name=key_by_name)
```

Comment:
@tpike3 can you explain why this would be needed?

Reply:
I figured out a way to avoid this by setting up a manual re-registration in meta-agents. so once I submit my update `agents.py` will not be touched. 

However, the reason it is needed can be seen in the warehouse `model.py` since `RobotAgent` is created in the initialization of the model there is no way to reference the object. (e.g. it can't beimported) So my challenge was how do we make this easy on the user to reference their agents, changing the key to a string in `agents_by_type` seemed to me the most intuitive.

Reply:
@quaquel I tried a few other ways to avoid this and while they would work it added a lot of unnecessary overheard of constantly rebuilding `agents_by_type`. 

If you have any ideas I am open to them. An example of the challenge cane be seen in the [warehouse model #L101](https://github.com/projectmesa/mesa-examples/pull/254/files#diff-b24aa2f9f2f9d709efa550fa103f10b9cbedb26061405595f653bfe04eaf04a2)

 As the `RobotAgent` is created dynamically it cant be referenced in the usual manner.

Reply:
removed